server/src/controllers/maproom/v2/cells/homeCell.ts

```diff
@@ -1,40 +1,68 @@
 import { Context } from "koa";
 import { User } from "../../../../models/user.model";
 import { ORMContext } from "../../../../server";
+import { WorldMapCell } from "../../../../models/worldmapcell.model";
+import { Save } from "../../../../models/save.model";
 
-export const homeCell = async (ctx: Context) => {
-  const user: User = ctx.authUser;
-  await ORMContext.em.populate(user, ["save"]);
+export const homeCell = async (ctx: Context, cell: WorldMapCell) => {
+  let user: User = ctx.authUser;
 
-  let save = user.save;
+
+  let save: Save;
+  let mine = user.userid === cell.uid;
+  user = await ORMContext.em.findOne(User, {
+    userid: cell.uid
+  })
+  save = await ORMContext.em.findOne(Save, {
+    baseid: cell.base_id.toString(),
+  })
 
   /**
    * For additional details about each of the following properties
    * visit the Wiki page: 'World Map v2: Cell Object'
    */
-  return {
-    uid: save.userid, // ToDo: Why do we have a userId for both user and save table? Fix
-    b: 2,
-    fbid: save.fbid,
-    pi: 0,
-    bid: save.homebaseid,
-    aid: 0,
-    i: 139,
-    mine: 1,
-    f: save.flinger,
-    c: save.catapult,
-    t: 0,
-    n: user.username,
-    fr: 0,
-    on: 0, // ToDo
-    p: save.protected,
-    r: save.resources,
-    m: save.monsters,
-    l: 1, // ToDo
-    d: save.destroyed,
-    lo: save.locked,
-    dm: save.damage,
-    pic_square: user.pic_square,
-    im: user.pic_square,
-  };
+
+  const online = isOnline(save.savetime);
+
+  let base = null;
+  if (save && user) {
+    base = {
+      uid: user.userid, // ToDo: Why do we have a userId for both user and save table? Fix
+      b: cell.base_type,
+      fbid: save.fbid,
+      pi: 0,
+      bid: save.baseid,
+      aid: 0,
+      i: 139,
+      mine: mine,
+      f: save.flinger,
+      c: save.catapult,
+      t: 0,
+      n: user.username,
+      fr: 0,
+      on: online, // ToDo
+      p: save.protected,
+      r: save.resources,
+      m: save.monsters,
+      l: 1, // ToDo
+      d: save.damage > 90,
+      lo: save.locked,
+      dm: save.damage,
+      pic_square: `https://api.dicebear.com/7.x/adventurer/png?backgroundType=solid&backgroundColor=b6e3f4,c0aede,d1d4f9&seed=${user.username}`,
```

Comment:
This is a pretty cool change thank you. Just thinking will we have CORS errors with this on prod. We can investigate later though

Reply:
[dicebear](https://www.dicebear.com/introduction/) allow all origin so i think it wouldn't be problem

![image](https://github.com/bym-refitted/backyard-monsters-refitted/assets/37240199/1838d507-9837-409b-9ea6-d5169a3efdda)